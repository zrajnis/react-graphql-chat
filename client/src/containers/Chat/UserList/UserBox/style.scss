.username,
.my-username {
  display: block;
  height: 16px;
  line-height: 16px;
  padding-bottom: 20px;
  width: 100%;
}

.username {
  color: $bg-white;
}

.my-username {
  color: $red;
}
