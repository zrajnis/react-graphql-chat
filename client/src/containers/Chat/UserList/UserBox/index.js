import React from 'react'
import PropTypes from 'prop-types'

import './style.scss'

const UserBox = ({ myUsername, username }) => (
  <li styleName={myUsername ? 'my-username' : 'username'}>
    {username}{myUsername && ' (You)'}
  </li>
)

UserBox.propTypes = {
  myUsername: PropTypes.bool.isRequired,
  username: PropTypes.string.isRequired
}

export default UserBox
