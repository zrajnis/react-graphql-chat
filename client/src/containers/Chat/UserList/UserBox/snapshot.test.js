import React from 'react'
import renderer from 'react-test-renderer'

import UserBox from 'containers/Chat/UserList/UserBox'

describe('User box component snapshot', () => {
  const username = 'testUser'

  it('renders the user box for a logged in user', () => {
    const tree = renderer.create(
      <UserBox
        myUsername
        username={username}
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders the user box for another user', () => {
    const tree = renderer.create(
      <UserBox
        myUsername={false}
        username={username}
      />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
