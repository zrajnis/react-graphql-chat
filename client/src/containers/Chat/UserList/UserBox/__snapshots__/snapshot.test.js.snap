// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User box component snapshot renders the user box for a logged in user 1`] = `
<li
  className="my-username--2An5F1qD"
>
  testUser
   (You)
</li>
`;

exports[`User box component snapshot renders the user box for another user 1`] = `
<li
  className="username--ThdCkMwR"
>
  testUser
</li>
`;
